subsys/net/ip/tcp2.c

```diff
@@ -832,6 +979,79 @@ static struct tcp *tcp_conn_new(struct net_pkt *pkt)
 	return conn;
 }
 
+static void update_acked_data(struct tcp *conn, struct tcphdr *th)
```

Comment:
In order to make this simpler, it could be further refactored into two sipler functions: one calculating how much data to remove from the beginning of the pending data and another one that removes it.